File: backend/internal/domain/evento.go
package domain

import "time"

type Visibilidade string

const (
	VisibilidadePublico    Visibilidade = "publico"
	VisibilidadeNaoListado Visibilidade = "nao_listado"
	VisibilidadePrivado    Visibilidade = "privado"
)

type TipoAcesso string

const (
	TipoAcessoIngresso TipoAcesso = "ingresso"
	TipoAcessoCadastro TipoAcesso = "cadastro"
)

type StatusEvento string

const (
	StatusEventoRascunho  StatusEvento = "rascunho"
	StatusEventoPublicado StatusEvento = "publicado"
	StatusEventoEncerrado StatusEvento = "encerrado"
	StatusEventoCancelado StatusEvento = "cancelado"
)

type ModoParticipantes string

const (
	ModoParticipantesNenhum          ModoParticipantes = "nenhum"
	ModoParticipantesConvite         ModoParticipantes = "convite"
	ModoParticipantesInscricaoAberta ModoParticipantes = "inscricao_aberta"
	ModoParticipantesAmbos           ModoParticipantes = "ambos"
)

// Evento segue a máquina de estados da seção 6 do plano:
// rascunho -> publicado -> encerrado; cancelado é terminal a partir de
// qualquer estado não-encerrado. "Vendas abertas" e "em andamento" são
// derivados das datas, não guardados aqui.
type Evento struct {
	ID                        int64 `gorm:"primaryKey"`
	OrganizadorID             int64
	LocalID                   *int64
	Titulo                    string
	Slug                      string
	Descricao                 string
	Categoria                 string
	CapaURL                   string
	InicioEm                  *time.Time
	FimEm                     *time.Time
	Timezone                  string
	ClassificacaoEtaria       string
	Visibilidade              Visibilidade
	TipoAcesso                TipoAcesso
	Status                    StatusEvento
	ModoParticipantes         ModoParticipantes
	InscricaoTalentosInicio   *time.Time
	InscricaoTalentosFim      *time.Time
	CapacidadeTotal           *int
	GarantiaHabilitada        bool
	AprovacaoManual           bool
	QRRotativo                bool
	ResultadoLiberadoEm       *time.Time
	PoliticaCancelamentoTexto string
	MaxItensPorPedido         int
	PublicadoEm               *time.Time
	CanceladoEm               *time.Time
	MotivoCancelamento        string
	CriadoEm                  time.Time
}

func (Evento) TableName() string {
	return "eventos"
}

File: frontend/src/lib/evento.ts
export type Evento = {
  id: number
  local_id: number | null
  titulo: string
  slug: string
  descricao: string
  categoria: string
  capa_url: string
  inicio_em: string | null
  fim_em: string | null
  timezone: string
  classificacao_etaria: string
  visibilidade: 'publico' | 'nao_listado' | 'privado'
  tipo_acesso: 'ingresso' | 'cadastro'
  status: 'rascunho' | 'publicado' | 'encerrado' | 'cancelado'
  modo_participantes: 'nenhum' | 'convite' | 'inscricao_aberta' | 'ambos'
  capacidade_total: number | null
  garantia_habilitada: boolean
  aprovacao_manual: boolean
  qr_rotativo: boolean
  cor_tema: string
  politica_cancelamento_texto: string
  max_itens_por_pedido: number
  publicado_em: string | null
}

export type TipoIngresso = {
  id: number
  nome: string
  descricao: string
  preco_centavos: number
  quantidade: number
  vendas_inicio: string | null
  vendas_fim: string | null
  min_por_pedido: number
  max_por_pedido: number
  ordem: number
  lote_grupo: string
  meia_entrada: boolean
  sessao_id: number | null
  ativo: boolean
}

export function formatarCentavos(centavos: number): string {
  return (centavos / 100).toLocaleString('pt-BR', { style: 'currency', currency: 'BRL' })
}
